crates/diem-rest-client/src/lib.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use diem_api_types::{MoveModuleBytecode, PendingTransaction, Transaction};
+use diem_client::{Response, State};
+use diem_crypto::HashValue;
+use diem_types::{account_address::AccountAddress, transaction::SignedTransaction};
+use move_core_types::{language_storage::StructTag, move_resource::MoveStructType};
+use reqwest::{header::CONTENT_TYPE, Client as ReqwestClient};
+use serde::Deserialize;
+use std::time::Duration;
+use url::Url;
+
+pub mod types;
+pub use types::{DiemAccount, Resource, RestError};
+
+const BCS_CONTENT_TYPE: &str = "application/x.diem.signed_transaction+bcs";
+const USER_AGENT: &str = concat!("diem-client-sdk-rust / ", env!("CARGO_PKG_VERSION"));
+
+#[derive(Clone, Debug)]
+pub struct Client {
+    inner: ReqwestClient,
+    base_url: Url,
+}
+
+impl Client {
+    pub fn new(base_url: Url) -> Self {
+        let inner = ReqwestClient::builder()
+            .timeout(Duration::from_secs(10))
```

Comment:
Let's make sure to follow up with an issue to make this adjustable.  10 seconds is a long timeout.